src/mongocxx/database.hpp

```diff
@@ -247,7 +247,9 @@ class MONGOCXX_API database {
     MONGOCXX_PRIVATE database(const class client& client, stdx::string_view name);
 
     class MONGOCXX_PRIVATE impl;
-    std::unique_ptr<impl> _impl;
+    MONGOCXX_PRIVATE std::unique_ptr<impl> _impl;
+    MONGOCXX_PRIVATE const impl& _get_impl() const;
```

Comment:
Methods before members, with a blank line in between. I think this would be ideal:

```
class MONGOCXX_PRIVATE impl;

MONGOCXX_PRIVATE std::unique_ptr<impl> _impl;
MONGOCXX_PRIVATE const impl& _get_impl() const;

std::unique_ptr<impl> _impl;
```